Restore automatic suppression of ID creation during org-refile and org-capture

mbhutton opened the pull request:
_(This description contains the same information as the commit messages, with added formatting)_

This PR contains 3 changes:

(1) Restore the previous behaviour of avoiding ID creation during `org-refile` and `org-capture` calls in the `org-bookmark-heading-make-record` function, by updating the detection mechanism in response to upstream org-mode changes.

The previous mechanism (added in 2018) checked for the `bookmark-name` variable being bound to certain values, but bookmark-name is no longer visible to this package, confirmed by debugging with edebug: `(boundp 'bookmark-name)` always returns `nil`.
The `org-refile.el` file was added upstream in 2020, with lexical binding, as a potential cause.

The new mechanism detects org-refile and org-capture calls using flags, which are toggled before and after the upstream calls from `org-refile` and `org-capture-store-last-position`.

The flags are intentionally set globally, beyond org-mode, as calls like org-capture are initiated outside of org-mode.

(2) When making bookmark records, always set the handler, even if not setting the ID.

Without this change, these org-bookmark-heading style records with no handler field cause odd behaviour when jumping to their bookmarks.

(3) Raise minimum required Emacs version from 24.4 to 26.1.

Suggested by Elisp linting, due to referencing the `org-capture-store-last-position` function.

mbhutton commented:
Thanks @alphapapa for this great package, it's really great to use and the code is really nice to work with :) 

I've written this PR to address #8 in a way that supports having both:
- (a) opt-in creation of IDs when manually creating bookmarks (using `org-bookmark-heading-make-ids`), as well as
- (b) the normal automatic creation of bookmarks for `org-refile-last-stored` etc, without the unwanted IDs.

The required bump to Emacs 26.1 isn't ideal, but even [Debian oldoldstable includes 26.1 now](https://packages.debian.org/search?keywords=emacs), as point of reference.

In terms of testing and auditing:
- The two methods being wrapped (`org-refile` and `org-capture-store-last-position `) are there with the same names in org from the Emacs 26.1 release date (May 2018) through to org HEAD
- I've tested with `org-bookmark-heading-make-ids` set to `t` and `nil`
- I've tested  bookmark-set, org-refile, and both types of org-capture (`c-c c-c` and `c-c c-w`)

alphapapa commented:
Hi @mbhutton,

Thanks for submitting this.  A few requests:

1. Please make PRs from feature branches, not your `master` branch.  Making PRs from master branches can cause problems later.
2. Please split the first commit into two commits and PRs, because the latter one can't be merged as-is, and the former probably can.  We can discuss feedback on the second one after it becomes a separate PR.
3. Updating the minimum Emacs version is probably fine and likely necessary, but I'll probably want to tag a stable release before changing that.

mbhutton commented:
@alphapapa Thanks!

I've split this into 2 PRs:
- fix-id-suppression bring the first one (1 commit), then
- fix-no-id-handler downstream of that (1 commit re always setting the handler)

I've removed the Emacs version bump.
